File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/logging-json/java/src/main/java/org/acme/logging/json/LoggingJsonExampleResource.java
package org.acme.logging.json;

import org.jboss.logging.Logger;
import org.jboss.resteasy.spi.NotImplementedYetException;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.ServerErrorException;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("/logging-json")
public class LoggingJsonExampleResource {

    private static final Logger LOG = Logger.getLogger(LoggingJsonExampleResource.class);

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        LOG.info("Reached LoggingJsonExampleResource hello() method");
        return "hello";
    }

    @GET
    @Path("/goodbye")
    @Produces(MediaType.TEXT_PLAIN)
    public String goodbye() {
        LOG.info("Reached LoggingJsonExampleResource goodbye() method, about to throw an Exception");
        throw new ServerErrorException(
                Response.Status.INTERNAL_SERVER_ERROR,
                new NotImplementedYetException("goodbye() not implemented yet"));
    }
}

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/examples/logging-json/kotlin/src/main/kotlin/org/acme/logging/json/LoggingJsonExampleResource.kt
package org.acme.logging.json

import org.jboss.logging.Logger
import org.jboss.resteasy.spi.NotImplementedYetException
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.ServerErrorException
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/logging-json")
class LoggingJsonExampleResource {
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello(): String {
        LOG.info("Reached LoggingJsonExampleResource hello() method")
        return "hello"
    }

    @GET
    @Path("/goodbye")
    @Produces(MediaType.TEXT_PLAIN)
    fun goodbye(): String {
        LOG.info("Reached LoggingJsonExampleResource goodbye() method, about to throw an Exception")
        throw ServerErrorException(
                Response.Status.INTERNAL_SERVER_ERROR,
                NotImplementedYetException("goodbye() not implemented yet"))
    }

    companion object {
        private val LOG = Logger.getLogger(LoggingJsonExampleResource::class.java)
    }
}